Fixing up a few issues in the new LLCP implementation

erbr-ot opened the pull request:
It was found that procedure pause handling was incomplete, so this is completed.
There was a slip in modification to struct llcp - missing member 'is_must_expire'
Finally it has turned out that the tx data pause/resume had a race condition, so the pause flag is changed into a counter, however without changing the API.

erbr-ot committed:
Bluetooth: controller: completing procedure pause handling

Implementing the final parts of procedure pause mechanism
This is needed to bar procedures from generating pdus during encryption
procedure handling

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

wip

erbr-ot committed:
Bluetooth: controller: fixing up struct ll_conn for new LLCP

Missing struct member added for non-legacy LLCP

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot committed:
Bluetooth: controller: fixing up possible race in data tx pause/resume

Turning the pause flag into a counter, to allow overlapping pause

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

thoh-ot commented on subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:
```diff
@@ -428,7 +449,7 @@ static void rr_st_idle(struct ll_conn *conn, uint8_t evt, void *param)
 				rr_act_run(conn);
 				rr_set_state(conn, RR_STATE_TERMINATE);
-			} else if (incompat == INCOMPAT_NO_COLLISION) {
+			} else if (!with_instant || incompat == INCOMPAT_NO_COLLISION) {
```
Isn't this change unrelated to the procedure pause and belongs in a separate commit regarding procedure collision?

erbr-ot replied:
yes it does - thank you!

erbr-ot committed:
Bluetooth: controller: fixing up problem in collision handling

Non-instant based procedures not properly handled, when instant
based procedure is ongoing

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot commented:
This is still pending reviews
